### PR TITLE
R4R:　Record/submit command and implementation

### PR DESCRIPTION
**Pull request by JinAirsOs**

### CODE REVIEW

**Comment by trevorfu** [comment]
this PR is related to issue: https://github.com/irisnet/irishub/issues/299

File: modules/record/msgs.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+//msg := record.NewMsgSubmitFile(filename, filepath, description, fileType, fromAddr, amount)
+func NewMsgSubmitFile(filename string, filepath string, description string, fileType string, proposer sdk.AccAddress, amount sdk.Coins, dataSize int64) MsgSubmitFile {
+	return MsgSubmitFile{
+		Filename:    filename,
+		Filepath:    filepath,
+		Description: description,
+		FileType:    fileType,
+		Proposer:    proposer,
+		Amount:      amount,
+		DataSize:    dataSize,
```
Comment (wukongcheng) [inline modules/record/msgs.go]: The blockchain should be able to record the ipfs hash of the uploaded file.

File: modules/record/errors.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+package record
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	DefaultCodespace sdk.CodespaceType = 5
```
Comment (wukongcheng) [inline modules/record/errors.go]: CodespaceType = 5  has been used by the gov module

File: modules/record/errors.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+	CodeUnknownProposal         sdk.CodeType = 1
+	CodeInactiveProposal        sdk.CodeType = 2
+	CodeAlreadyActiveProposal   sdk.CodeType = 3
+	CodeAlreadyFinishedProposal sdk.CodeType = 4
+	CodeAddressNotStaked        sdk.CodeType = 5
```
Comment (wukongcheng) [inline modules/record/errors.go]: Do we need to handle the errors about proposal?

File: modules/record/errors.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	CodeInvalidFilename         sdk.CodeType = 6
+	CodeInvalidDescription      sdk.CodeType = 7
+	CodeInvalidProposalType     sdk.CodeType = 8
+	CodeInvalidVote             sdk.CodeType = 9
+	CodeInvalidGenesis          sdk.CodeType = 10
+	CodeInvalidProposalStatus   sdk.CodeType = 11
+)
+
+func ErrInvalidFilename(codespace sdk.CodespaceType, title string) sdk.Error {
+	return sdk.NewError(codespace, CodeInvalidFilename, fmt.Sprintf("Proposal Title '%s' is not valid", title))
```
Comment (wukongcheng) [inline modules/record/errors.go]: The error msg is not right

File: modules/record/handler.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+	//todo
+	/*proposal := keeper.NewProposal(ctx, msg.Title, msg.Description, msg.ProposalType, msg.Param)
+
+	err, votingStarted := keeper.AddDeposit(ctx, proposal.GetProposalID(), msg.Proposer, msg.Amount)
+	if err != nil {
+		return err.Result()
+	}
+
+	proposalIDBytes := keeper.cdc.MustMarshalBinaryBare(proposal.GetProposalID())*/
```
Comment (wukongcheng) [inline modules/record/handler.go]: remove these useless code coped from gov

File: modules/record/keeper.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/irisnet/irishub/modules/iparams"
+)
+
+// nolint
+
+// Record Keeper
+type Keeper struct {
+	// The reference to the ParamSetter to get and set Global Params
+	ps iparams.GovSetter
```
Comment (wukongcheng) [inline modules/record/keeper.go]: iparams.GovSetter has been refactored

File: modules/record/keeper.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+	// The reference to the CoinKeeper to modify balances
+	ck bank.Keeper
+
+	// The ValidatorSet to get information about validators
+	vs sdk.ValidatorSet
+
+	// The reference to the DelegationSet to get information about delegators
+	ds sdk.DelegationSet
+
```
Comment (wukongcheng) [inline modules/record/keeper.go]: Do we need the ValidatorSet & DelegationSet ?

File: modules/record/msgs.go [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+const MsgType = "record"
+
+//-----------------------------------------------------------
+// MsgSubmitFile
+type MsgSubmitFile struct {
+	Filename    string         //  Filename of the File
+	Filepath    string         //  full path of the File
+	Description string         //  Description of the File
+	FileType    string         //  Type of file
+	Proposer    sdk.AccAddress //  Address of the proposer
```
Comment (wukongcheng) [inline modules/record/msgs.go]: Why Proposer? The one who submit the file?

**Pull request merged**